caikit/core/data_model/dataobject.py

```diff
@@ -67,16 +67,11 @@ class will be the raw input representation of a @dataclass
 
         # Get the annotations that will go into the dataclass
         if name != "DataObjectBase":
-            raw_field_names = attrs.get("__annotations__")
-            if raw_field_names is None:
+            field_names = attrs.get("__annotations__")
+            if field_names is None:
                 raise TypeError(
                     "All DataObjectBase classes must follow dataclass syntax"
                 )
-
-            # TODO: Sort out oneof field names
```

Comment:
This ended up being accomplished through guaranteeing that all `DataBase` derived classes stored oneof data by the name of the oneof rather than the name of the field, so nothing more was needed here.